numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,133 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def sliding_window_view(x, shape, axis=None, subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+
+    axis : int or tuple of int, optional
+        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `shape[i]` will refer to axis `i` of `x`.
```

Comment:
```suggestion
    shape : int or tuple of int
        The shape of the window. If `axis` is not present, must have same
        length as the number of input array dimensions.
        Single integers `i` are treated as if they were the tuple `(i,)`.

    axis : int or tuple of int, optional
        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
        otherwise `shape[i]` will refer to axis `i` of `x`.
        Single integers `i` are treated as if they were the tuple `(i,)`.
```

Reply:
Good clarification, thanks!